### PR TITLE
Docs: Fix link to developing plugins

### LINKED ISSUE #75788: 404 - page not found error on opening grafana/developers /plugin-tools in CONTRIBUTING.md
### What happened?

Screenshot 2023-10-01 at 9 55 27 AM
Screenshot 2023-10-01 at 9 54 15 AM

### What did you expect to happen?

Page should load plugin-tools

### Did this work before?

Yes

### How do we reproduce it?

1. Direct to this URL - https://github.com/grafana/grafana
2. Under the Contributing section, click on Contributing guide
3. Redirect to - **https://github.com/grafana/grafana/blob/HEAD/CONTRIBUTING.md**
4. Head to bottom of the page and under "Where do I go from here" section click on developing plugins.
5. Now, the redirected page shows 404 error.

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana:
OS:
Browser:

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by shaviz-s**
**What is this feature?**

Fixed URL to developing plugins redirect. Page should now load plugin-tools.

**Why do we need this feature?**

Currently the link redirects to a 404 page. This update fixes that.

**Who is this feature for?**

Any user reading the documentations.

**Which issue(s) does this PR fix?**:

Fixes #75788

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by chri2547** [comment]
@josmperez , can you please take a look at this PR that corrects a number of links to your docs. I wonder if you have already corrected these? Thank you!

**Review by josmperez — APPROVED** [review]
LGTM, thanks.

**Comment by jdbaldry** [comment]
Thank you for contributing!

The only URL I think we need to fix here is the one in `CONTRIBUTING.md` which definitely needs the fully qualified URL in order to take readers to the correct page.

For all the pages in `docs/sources`, they are used as part of a Hugo site build and are not necessarily intended to resolve correctly in GitHub. 

The use of partial URLs work the same as the fully qualified URLs on the site. We previously required the use of partial URLs so that links worked correctly on the staging site which does not share the grafana.com hostname. However, we have since made changes on the site side that means this is no longer required. `doc-validator` is enforcing an old rule on some parts of the repository. I will update the code today and perform a new release so that future PRs are not blocked on the out-of-date linting error.

File: docs/sources/administration/data-source-management/index.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ You can assign data source permissions to users, teams, and roles which will all
 
 When you enable query and resource caching, Grafana temporarily stores the results of data source queries and resource requests. When you or another user submit the same query or resource request again, the results will come back from the cache instead of from the data source.
 
-When using Grafana, a query pertains to a request for data frames to be modified or displayed. A resource relates to any HTTP requests made by a plugin, such as the Amazon Timestream plugin requesting a list of available databases from AWS. For more information on data source queries and resources, please see the developers page on [backend plugins](/developers/plugin-tools/introduction/backend-plugins).
+When using Grafana, a query pertains to a request for data frames to be modified or displayed. A resource relates to any HTTP requests made by a plugin, such as the Amazon Timestream plugin requesting a list of available databases from AWS. For more information on data source queries and resources, please see the developers page on [backend plugins](https://grafana.com/developers/plugin-tools/introduction/backend-plugins).
```
Comment (chri2547) [inline docs/sources/administration/data-source-management/index.md:92]:
```suggestion
When using Grafana, a query pertains to a request for data frames to be modified or displayed. A resource relates to any HTTP requests made by a plugin, such as the Amazon Timestream plugin requesting a list of available databases from AWS. For more information on data source queries and resources, please see the developers page on [backend plugins](/developers/plugin-tools/introduction/backend-plugins).
```

File: docs/sources/administration/data-source-management/index.md [hunk 2]
```diff
@@ -124,7 +124,7 @@ Query caching works for Grafana's [built-in data sources]({{< relref "../../data
 To verify that a data source works with query caching, follow the [instructions below](#enable-and-configure-query-caching) to **Enable and Configure query caching**. If caching is enabled in Grafana but the Caching tab is not visible for the given data source, then query caching is not available for that data source.
 
 {{% admonition type="note" %}}
-Some data sources, such as Elasticsearch, Prometheus, and Loki, cache queries themselves, so Grafana _query_ caching does not significantly improve performance. However, _resource_ caching may help. See the developers page on [plugin resources](/developers/plugin-tools/introduction/backend-plugins) for details.
+Some data sources, such as Elasticsearch, Prometheus, and Loki, cache queries themselves, so Grafana _query_ caching does not significantly improve performance. However, _resource_ caching may help. See the developers page on [plugin resources](https://grafana.com/developers/plugin-tools/introduction/backend-plugins) for details.
```
Comment (chri2547) [inline docs/sources/administration/data-source-management/index.md:127]:
```suggestion
Some data sources, such as Elasticsearch, Prometheus, and Loki, cache queries themselves, so Grafana _query_ caching does not significantly improve performance. However, _resource_ caching may help. See the developers page on [plugin resources](/developers/plugin-tools/introduction/backend-plugins) for details.
```

File: docs/sources/administration/data-source-management/index.md [hunk 3]
```diff
@@ -195,4 +195,4 @@ For more documentation on a specific data source plugin's features, including it
 
 ### Create a data source plugin
 
-To build your own data source plugin, refer to the ["Build a data source plugin"](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building a plugin](/developers/plugins-tools).
+To build your own data source plugin, refer to the ["Build a data source plugin"](https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building a plugin](https://grafana.com/developers/plugin-tools).
```
Comment (chri2547) [inline docs/sources/administration/data-source-management/index.md:198]:
```suggestion
To build your own data source plugin, refer to the ["Build a data source plugin"](/developers/plugin-tools/tutorials/build-a-data-source-plugin) tutorial and our documentation about [building a plugin](/developers/plugin-tools).
```

File: docs/sources/administration/plugin-management/index.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ weight: 600
 
 Besides the wide range of visualizations and data sources that are available immediately after you install Grafana, you can extend your Grafana experience with _plugins_.
 
-You can [install](#install-a-plugin) one of the plugins built by the Grafana community, or [build one yourself](/developers/plugin-tools).
+You can [install](#install-a-plugin) one of the plugins built by the Grafana community, or [build one yourself](https://grafana.com/developers/plugin-tools).
```
Comment (chri2547) [inline docs/sources/administration/plugin-management/index.md:20]:
```suggestion
You can [install](#install-a-plugin) one of the plugins built by the Grafana community, or [build one yourself](/developers/plugin-tools).
```

File: docs/sources/datasources/_index.md [hunk 5]
```diff
@@ -29,7 +29,8 @@ After you add and configure a data source, you can use it as an input for many o
 This documentation describes how to manage data sources in general,
 and how to configure or query the built-in data sources.
 For other data sources, refer to the list of [datasource plugins](/grafana/plugins/).
-To develop a custom plugin, refer to our [plugin development documentation](/developers/plugin-tools).
+
+To develop a custom plugin, refer to [Build a plugin](https://grafana.com/developers/plugin-tools)
```
Comment (chri2547) [inline docs/sources/datasources/_index.md:33]:
```suggestion
To develop a custom plugin, refer to [Build a plugin](/developers/plugin-tools)
```

File: docs/sources/developers/_index.md [hunk 6]
```diff
@@ -13,8 +13,8 @@ weight: 190
 
 Go to the [Grafana developer portal](/developers) to access the following documentation:
 
-- [Grafana plugin development](/developers/plugin-tools)
-- [Grafana UI component library](https://developers.grafana.com)
+- [Grafana plugin development](https://grafana.com/developers/plugin-tools)
```
Comment (chri2547) [inline docs/sources/developers/_index.md:16]:
```suggestion
- [Grafana plugin development](/developers/plugin-tools)
```

File: docs/sources/fundamentals/dashboards-overview/index.md [hunk 7]
```diff
@@ -43,7 +43,7 @@ A Grafana plugin is software that adds new capabilities to Grafana. They come in
 
 The data coming into the plugin from the data source might be in many different formats (such as JSON, rows and columns, or CSV), but when it leaves the plugin and moves through the rest of the gates toward a visualization, it's always in data frames.
 
-Currently, Grafana offers a diverse range of 155 data sources that you can use. The most commonly used options are already pre-installed and accessible. Before exploring other options, look for an existing data source that matches your requirements. Grafana constantly updates the list, but if you don't find a suitable data source, you can browse through the [plugin catalog](/grafana/plugins/?type=datasource) or [create a plugin](/developers/plugin-tools).
+Currently, Grafana offers a diverse range of 155 data sources that you can use. The most commonly used options are already pre-installed and accessible. Before exploring other options, look for an existing data source that matches your requirements. Grafana constantly updates the list, but if you don't find a suitable data source, you can browse through the [plugin catalog](/grafana/plugins/?type=datasource) or [create a plugin](https://grafana.com/developers/plugin-tools).
```
Comment (chri2547) [inline docs/sources/fundamentals/dashboards-overview/index.md:46]:
```suggestion
Currently, Grafana offers a diverse range of 155 data sources that you can use. The most commonly used options are already pre-installed and accessible. Before exploring other options, look for an existing data source that matches your requirements. Grafana constantly updates the list, but if you don't find a suitable data source, you can browse through the [plugin catalog](/grafana/plugins/?type=datasource) or [create a plugin](/developers/plugin-tools).
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -83,5 +83,5 @@ Before we can accept your pull request, you need to [sign our CLA](https://grafa
 
 - Set up your [development environment](contribute/developer-guide.md).
 - Learn how to [contribute to our documentation](contribute/documentation/README.md).
-- Get started [developing plugins](/developers/plugin-tools) for Grafana.
+- Get started [developing plugins](https://grafana.com/developers/plugin-tools) for Grafana.
```
Comment (chri2547) [inline CONTRIBUTING.md:86]:
```suggestion
- Get started [developing plugins](/developers/plugin-tools) for Grafana.
```

Comment (jdbaldry) [reply CONTRIBUTING.md:86]: This is the one we definitely want to keep.

**Review by chri2547 — APPROVED** [review]

**Pull request merged**